packages/headless/src/features/triggers/triggers-slice.ts

```diff
@@ -4,18 +4,32 @@ import {getTriggerInitialState} from './triggers-state';
 import {
   Trigger,
   TriggerRedirect,
-  isTriggerRedirect,
+  isRedirectTrigger,
+  isQueryTrigger,
+  TriggerQuery,
 } from './../../api/search/trigger';
+import {clearQueryTrigger} from './trigger-actions';
 
 export const triggerReducer = createReducer(
   getTriggerInitialState(),
   (builder) =>
-    builder.addCase(executeSearch.fulfilled, (state, action) => {
-      const redirectTriggers: Trigger[] = action.payload.response.triggers.filter(
-        (trigger) => isTriggerRedirect(trigger)
-      );
-      state.redirectTo = redirectTriggers.length
-        ? (redirectTriggers[0] as TriggerRedirect).content
-        : '';
-    })
+    builder
+      .addCase(executeSearch.fulfilled, (state, action) => {
```

Comment:
If the initial states for redirectTo / query is `null` it should return to `null` when there are no request. I can see a potential problem for this state to automalically from `null` to `''` (empty string) as soon as a search is trigger

Reply:
If you prefer, you can also set the initial state to `''` and remove the `null` definition

Reply:
Yes that will be neater.